plugins/vpp/l3plugin/route_config_test.go

```diff
@@ -0,0 +1,426 @@
+		Weight:            6,
+		OutgoingInterface: "tap1",
+	}, "0")
+	Expect(err).ShouldNot(HaveOccurred())
+
+	routes := plugin.GetRouteIndexes().GetMapping().ListNames()
+	Expect(routes).To(HaveLen(1))
+	Expect(routes[0]).To(Equal("vrf0-fd21:dead:abcd::/48-fd21:cdef:dead::"))
+}
+
+// Test modify of cached route
+func TestModifyCachedRoute(t *testing.T) {
+	// Setup
+	ctx, connection, plugin, _ := routeTestSetup(t)
+	defer routeTestTeardown(connection, plugin)
+
+	err := plugin.ConfigureRoute(&l3.StaticRoutes_Route{
+		VrfId:             0,
+		DstIpAddr:         "10.1.1.3/32",
+		NextHopAddr:       "192.168.1.13",
+		Weight:            6,
+		OutgoingInterface: "tap1",
+	}, "0")
+	Expect(err).ShouldNot(HaveOccurred())
+	routes := plugin.GetCachedRouteIndexes().GetMapping().ListNames()
+	Expect(routes).To(HaveLen(1))
+	Expect(routes[0]).To(Equal("vrf0-10.1.1.3/32-192.168.1.13"))
+
+	ctx.MockVpp.MockReply(&ip.IPAddDelRouteReply{})
+	ctx.MockVpp.MockReply(&ip.IPAddDelRouteReply{})
+	err = plugin.ModifyRoute(&l3.StaticRoutes_Route{
+		VrfId:             0,
+		DstIpAddr:         "fd21:dead:abcd::/48",
+		NextHopAddr:       "fd21:cdef:dead::",
+		Weight:            6,
+		OutgoingInterface: "tap1",
+	}, &l3.StaticRoutes_Route{
+		VrfId:             0,
+		DstIpAddr:         "10.1.1.3/32",
+		NextHopAddr:       "192.168.1.13",
+		Weight:            6,
+		OutgoingInterface: "tap1",
+	}, "0")
+	Expect(err).Should(HaveOccurred())
+
+	Expect(plugin.GetRouteIndexes().GetMapping().ListNames()).To(HaveLen(0))
+	routes = plugin.GetCachedRouteIndexes().GetMapping().ListNames()
+	// TODO : uncomment after LUNAR-1440 fix
+	//Expect(routes).To(HaveLen(1))
+	//Expect(routes[0]).To(Equal("vrf0-fd21:dead:abcd::/48-fd21:cdef:dead::"))
+}
+
+// Test modify of cached route
+func TestModifyCachedRouteInterface(t *testing.T) {
+	// Setup
+	ctx, connection, plugin, ifIndexes := routeTestSetup(t)
+	defer routeTestTeardown(connection, plugin)
+
+	// add interface
+	ifIndexes.GetMapping().RegisterName("tap2", 1, nil)
+
+	err := plugin.ConfigureRoute(&l3.StaticRoutes_Route{
+		VrfId:             0,
+		DstIpAddr:         "10.1.1.3/32",
+		NextHopAddr:       "192.168.1.13",
+		Weight:            6,
+		OutgoingInterface: "tap1",
+	}, "0")
+	Expect(err).ShouldNot(HaveOccurred())
+	routes := plugin.GetCachedRouteIndexes().GetMapping().ListNames()
+	Expect(routes).To(HaveLen(1))
+	Expect(routes[0]).To(Equal("vrf0-10.1.1.3/32-192.168.1.13"))
+
+	ctx.MockVpp.MockReply(&ip.IPAddDelRouteReply{})
+	ctx.MockVpp.MockReply(&ip.IPAddDelRouteReply{})
+	err = plugin.ModifyRoute(&l3.StaticRoutes_Route{
+		VrfId:             0,
+		DstIpAddr:         "fd21:dead:abcd::/48",
+		NextHopAddr:       "fd21:cdef:dead::",
+		Weight:            6,
+		OutgoingInterface: "tap2",
+	}, &l3.StaticRoutes_Route{
+		VrfId:             0,
+		DstIpAddr:         "10.1.1.3/32",
+		NextHopAddr:       "192.168.1.13",
+		Weight:            6,
+		OutgoingInterface: "tap1",
+	}, "0")
+	// TODO : uncomment after LUNAR-1440 fix
```

Comment:
This can now be enabled.

Reply:
Yes, I was waiting till fix will merged.